asmcomp/arm64/emit.mlp

```diff
@@ -274,7 +274,7 @@ let emit_stack_adjustment n =
 
 let output_epilogue env f =
   let n = frame_size env in
-  if env.f.fun_contains_calls then
+  if env.f.fun_frame_required then
```

Comment:
Looking at this and other replacements of `fun_contains_calls` with `fun_frame_required` in isolation, it is not immediately clear to me why this change is correct. In particular, `fun_frame_required` on Arm64 is the one defined in `asmcomp/stackframegen.ml`, which is 

```
method frame_required f contains_calls =
  contains_calls || 
  f.fun_num_stack_slots.(0) > 0 || f.fun_num_stack_slots.(1) > 0
```
which is weaker than `fun_contains_calls`? What am I missing?

Reply:
In what sense is it weaker?  Notice that `frame_required` will always return `true` if `contains_calls` is true.

Reply:
Oh, by "weaker" you mean "frame_required is true even for some functions that do not contain calls".  This is true, but as I wrote above:
_However, the current code is not taking advantage of [functions that need a frame but don't contain calls], always saving the return address if a stack frame is set up. I don't think it matters performance-wise, so I propose to simplify the code accordingly._

Reply:
My reasoning is that a leaf function (containing no calls) that cannot run entirely in registers and requires a stack frame must be pretty big already, so the cost of saving & restoring the return address is negligible.

Reply:
Thanks for the clarification. The comment does indeed make sense now that you point it out again :-)